modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/util/NavigationItem.java

```diff
@@ -17,6 +17,18 @@
  */
 public class NavigationItem extends HashMap<String, Object> {
 
+	public void putData(String key, Boolean value) {
```

Comment:
Can we just change existing 
```java
putData(String key, String value)
```
into
```java
putData(String key, Object value)
```
It ends up in `Map<String, Object> data` anyways, so there is really no reason not to.

I don't think this will break existing API. It would not break anything in runtime, but I'm not sure about compile time :thinking:

Reply:
Let's try

Reply:
Java :clown_face: 

Then, to avoid code duplication, I believe you can do the main implantation in
```java
putData(String key, Object value)
```
and then have something like
```java
putData(String key, String value) {
    Object objectValue = value;

    putData(key, objectValue);
}
```
There might be a more elegant way to do this, I'm not sure.

Reply:
That seems to work :rocket: